File: forth/pe001.fs
#! /usr/bin/gforth
: mod5 ( n -- ismod )
5 mod 0= ;

: mod3 ( n -- ismod )
3 mod 0= ;

: add-number ( sum n -- sum n )
dup rot + swap ;

: test-number ( n -- )
dup mod5 if add-number else 
dup mod3 if add-number then then drop ;

: test-loop ( n  -- )
0 do i test-number loop ;

: pe001 ( -- )
0 1000 test-loop . cr ;

pe001
bye

File: typescript/pe001.ts
function pe001(imax01:number): number {
  let total:number = 0

  for (var i=0; i < imax01; i++)
    if (!(i % 3) || !(i % 5))
      total += i  
  return total
}

const imax01:number = 1000
console.log(pe001(imax01))
